accepted/0000-pure-mode.md

```diff
@@ -0,0 +1,344 @@
+# Pure mode
```

Comment:
i don't think "pure" is the right term to use here

Reply:
I updated the reasoning for choosing the term `pure-mode`. Please have a look, and if you don't agree please explain why.

Reply:
I understand the analogy, and obviously this is pure semantics.

* pure can have negative cultural and historic connotations related to race and religion
* pure implies a value judgement
* the pure function analogy isn't immediately obvious to the reader and wouldn't significantly preload accurate assumptions if they did

Reply:
Makes sense, I guess we have to find a better name then.

Reply:
Simply "unhoisted mode" might be sufficient.

Reply:
@fritzy I feel like that could have a negative impact on user perception compared to a unique name. Why _wouldn't_ I want my packages hoisted? That sounds like a good thing. Further, "unhoisted" makes the addition of new modes later harder since you'll have hoisted, unhoisted, and <x> where <x> is also conceptually "unhoisted" but that name was already taken.

IMO it's better to not have a negative prefix to describe what this is doing but instead have a direct name.

+1 to "isolated" - of the options here, it feels like the least loaded ("non-flat" suffers from a negative prefix with a new term for a concept that's already defined, "strict mode" is already a JavaScript thing, and "predictable" does have an assertion of value)

Reply:
I will throw another idea: hidden-subdependencies

Reply:
A poll was created to vote on a name